File: Baekjoon/Python/누적합/11659_구간_합_구하기_4/11659.py
import sys

input = sys.stdin.readline

N, M = map(int, input().strip().split())

my_list = list(map(int, input().strip().split()))

for i in range(1, N):
    my_list[i] = my_list[i] + my_list[i-1]

for _ in range(M):
    start, end = map(int, input().strip().split())
    if start == 1:
        print(my_list[end-1])
    else:
        print(my_list[end-1]-my_list[start-2])

File: Baekjoon/JavaScript/부분합/11659_부분합/11659.js
let [command, numbers, ...input] = require("fs")
  // .readFileSync("/dev/stdin")
  .readFileSync("input.txt")
  .toString()
  .trim()
  .split("\n")

const [N, M] = command.split(" ").map((x) => parseInt(x))

const ans = []

numbers = numbers.split(" ").map((x) => parseInt(x))

for (let i = 0; i < M; i++) {
  const [start, end] = input[i].split(" ")
  let cum = 0
  for (const j of numbers) {
    cum += j
  }
  console.log(cum)
}
